app/scripts/analysis/charts/charts.coffee

```diff
@@ -1,86 +1,671 @@
+        console.log 'charts init called'
 
       destroy: () ->
 
-      msgList: msgList
-]
+      msgList: _msgList
+])
 
-###
-  @description: Manager for all communication b/w modules.
-    Only this service inside this module, has access to sandbox.
-  @type: service
-###
-charts.service 'app_analysis_charts_eventMngr', [
-  'app_analysis_charts_compute'
-  (app_analysis_charts_compute)->
-    sb = null
-
-    msgList =
-      outcome: ['chart plotted']
-      income:
-        'plot chart':
-          method: app_analysis_charts_compute
-          outcome: 'chart plotted'
+.factory( 'app_analysis_charts_manager', [
+  ()->
+    _sb = null
+
+    _msgList =
+      outgoing: ['get table']
+      incoming: ['take table']
       scope: ['charts']
 
-    eventManager = (msg, data) ->
-      try
-        _data = msgList.income[msg].method.apply null,data
-        #last item in data is a promise.
-        data[data.length - 1].resolve _data if _data isnt false
-      catch e
-        console.log e.message
-        alert 'error in database'
-
-      sb.publish
-        msg: msgList.income[msg].outcome
-        data: _data
-        msgScope: msgList.scope
-
-    setSb: (_sb) ->
-      return false if _sb is undefined
-      sb = _sb
-
-    getMsgList: () ->
-      msgList
-
-    listenToIncomeEvents: () ->
-      app_analysis_charts_compute()
-      for msg of msgList.income
-        console.log 'subscribed for ' + msg
-        sb.subscribe
-          msg: msg
-          listener: eventManager
-          msgScope: msgList.scope
-          context: console
-]
+    _setSb = (sb) ->
+      _sb = sb
 
-charts.factory 'app_analysis_charts_compute',[
-    '$q'
-    ($q)->
-        (args...)->
-            #check for tname.
+    _getSb = () ->
+      _sb
 
-        #console.log '%c chartsMainCtrl:'+typeof chartsMainCtrl , 'color:yellow'
+    _getMsgList = () ->
+      _msgList
 
-]
+    getSb: _getSb
+    setSb: _setSb
+    getMsgList: _getMsgList
+])
+
+.controller('mainchartsCtrl', [
+  'app_analysis_charts_manager'
+  '$scope'
+  (ctrlMngr,$scope) ->
+    console.log 'mainchartsCtrl executed'
+    _chart_data = null
+
+    _updateData = () ->
+      $scope.chartData = _chart_data
+
+    $scope.$on 'charts:graphDiv', (event, data) ->
+      _chart_data = data
+      console.log data
+      _updateData()
+])
+
+
+
+.controller('sidechartsCtrl',[
+  'app_analysis_charts_manager'
+  '$scope'
+  '$rootScope'
+  '$stateParams'
+  '$q'
+  'app_analysis_charts_dataTransform'
+  (ctrlMngr, $scope, $rootScope, $stateParams, $q, dataTransform) ->
+    console.log 'sidechartsCtrl executed'
+    _chartData = null
+    _headers = null
+
+    $scope.selector1={}
+    $scope.selector2={}
+    $scope.selector3={}
+    $scope.graphInfo = {graph:"", x:"", y:"",z:""}
+    $scope.graphs = [{name:'Bar Graph', value:0},{name:'Scatter Plot', value:1},{name:'Histogram', value:2},{name:'Bubble Chart', value:3},{name:'Pie Chart', value:4}]
```

Comment:
(style): array of objects should be defined in CoffeeScript style for better readability, e.g. see [app.coffee](https://github.com/nponarul/SOCRAT/blob/charts/app/app.coffee#L149); also, there should be space after semicolon, but not before.